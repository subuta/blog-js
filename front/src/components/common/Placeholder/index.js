import React from 'react'

import classes from './style'

export default ({style = {}}) => (
  <div className={classes.Placeholder} style={style}/>
)
