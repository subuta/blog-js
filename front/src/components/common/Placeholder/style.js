import createWithStyles from 'src/utils/style'

const Placeholder = {
  display: 'inline-block',
  backgroundColor: 'currentColor',
  borderRadius: 20,
  height: 12,
  minWidth: 12,
  width: '100%',
  lineHeight: 1
}

export default createWithStyles({
  Placeholder
})
