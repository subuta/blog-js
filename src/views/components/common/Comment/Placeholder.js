import React from 'react'
import MarkdownContent from 'src/views/components/common/MarkdownContent'

import withStyles from './style'

import Avatar from 'src/views/components/common/Avatar'

export default withStyles((props) => {
  const {
    styles,
    style,
    width,
    variation = 'comment'
  } = props

  let commentWrapperClass = styles.CommentWrapper

  return (
    <div
      className={commentWrapperClass}
      style={style}
    >
      <Avatar avatar={null} nickname={null}/>

      <div className={styles.CommentPlaceholder}>
        <div className={styles.Nickname}>
          <span className="nickname"/>

          <small className="commentedAt"/>
        </div>

        <div className={variation}
             style={{width}}
        />
      </div>
    </div>
  )
})
