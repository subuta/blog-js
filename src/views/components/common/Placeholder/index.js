import React from 'react'

import withStyles from './style'

export default withStyles(({styles, style = {}}) => (
  <div className={styles.Placeholder} style={style} />
))
