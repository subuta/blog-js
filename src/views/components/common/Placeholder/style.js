import createWithStyles from 'src/views/utils/style'

const Placeholder = {
  display: 'inline-block',
  backgroundColor: 'currentColor',
  borderRadius: 20,
  height: 12,
  minWidth: 12,
  width: '100%',
  lineHeight: 1
}

export default createWithStyles({
  Placeholder
})
