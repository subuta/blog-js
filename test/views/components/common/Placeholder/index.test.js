/* global describe, it */
import React from 'react'
import { create } from 'react-test-renderer'

import Placeholder from 'src/components/common/Placeholder'

describe('Placeholder', () => {
  it('should be rendered correctly', () => {
    const tree = create(
      <Placeholder />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
