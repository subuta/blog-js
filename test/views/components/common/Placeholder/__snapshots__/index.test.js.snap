// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Placeholder should be rendered correctly 1`] = `
<div
  className="fpkkk6o"
  style={Object {}}
/>
`;
